challenges/systems-intern/worker-library.tex

```diff
@@ -0,0 +1,205 @@
+    \item Unstructured code. We've seen candidates leaving commented chunks of code, having one large file with all the code, not having code structure at all.
+    \item Not communicating. Some candidates have submitted all their code to the master branch without raising pull requests, which does not give us the ability to provide feedback on the various implementation phases. We are a distributed team, so structured, asynchronous communication is critical to us.
+    \item Implementing custom security algorithms/authentication schemes is always a bad idea unless you are a trained security researcher/engineer. It is definitely a bad idea for this task - try to stick to industry proven security methods as much as possible.
+    \end{itemize}
+
+\subsection{Scoring}
+    
+We want to be as transparent as possible on how we will be scoring your submission.
+The following table provides a description of different areas you will be evaluated on and how they will affect your overall score.
+
+\begin{center}
+\rowcolors{2}{gray!80!gray!50}{gray!50!gray!20}
+\begin{tabular}{ | m{25em} | m{5em}| m{5em} | }
+  \hline
+  \rowcolor{blue!60!black!10}
+  Description & Points Awarded & Points Subtracted \\
+  \hline
+  The submitted code has a clear and modular structure & +1 & -1 \\
+  \hline   
+  The candidate communicated their progress during the interview & +1 & -1 \\
+  \hline
+  The program builds are reproducible & +1 & -1 \\
+  \hline
+  README provides clear instructions & +1 & -1 \\
+  \hline
+  The candidate outlined the key design points in the design document & +1 & -1 \\
+  \hline
+  The code has no obvious data races and deadlocks & +1 & -1 \\
+  \hline
+  The code provides examples of tests covering key components & +1 & -1 \\
+  \hline
+  The code provides clear error handling and reporting & +1 & -1 \\
+  \hline
+  The program is working according to the specification & +1 & -1 \\
+  \hline
+  The candidate demonstrates ability to handle and apply feedback & +1 & -1 \\
+  \hline
+\end{tabular}
+\end{center}
+
+\subsection{Scoring}
+
+It is OK to ask the interview team questions. Some folks stay away from
+asking questions to avoid appearing less experienced, so we provide examples of questions
+to ask and questions we expect candidates to figure out on their own.
+
+Here is a great question to ask:
+
+``Is it OK to discard the output of a process that exceeds a predefined hardcoded value?.''
+
+It demonstrates that you thought about this problem domain, recognize the trade off and are saving you and the team time by not implementing it.
+
+This is the question we expect candidates to figure out on their own:
+
+``What version of Go should I use? What dependency manager should I use?''
+
+Unless specified in the requirements, pick the solution that works best for you.
+
+\section{Tools}
+
+This task should be implemented in Go, C++, Rust or Java and should work on 64-bit Linux machines.
```

Comment:
Didn't we remove Java from our allowed languages, or is that ok specifically for the interns?

Reply:
That's specifically for internships